### PR TITLE
[JS] feat: add helper functions to to get Teams channels, members, and details

### PR DESCRIPTION
**Pull request by yiqing-zhao**
## Linked issues

closes: #minor

## Details

Provide a list of your changes here. If you are fixing a bug, please provide steps to reproduce the bug.

#### Change details

> We have encapsulated three APIs from [`TeamsInfo`](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder/src/teamsInfo.ts) class of `botbuilder` library: `getTeamChannels`, `getTeamDetails`, and `getPagedMember`, to facilitate easier proactive message sending. These three APIs have been wrapped as an initial sample. If this approach is deemed acceptable, we intend to extend this encapsulation to include additional APIs provided by `TeamsInfo`, such as `sendMessageToListOfUsers`, `sendMessageToAllUsersInTenant`, and more.

PM Contact: @MuyangAmigo 

**code snippets**:
To use the APIs:
```ts
for (const reference of conversationReferences) {
  await app.sendProactiveActivity(
    reference,
    "Proactive message from RestifyNotiBot!"
  );

  if (reference.conversation?.conversationType === "channel") {
    const details = await app.getTeamDetails(reference);
    const teamMembers = await app.getPagedMembers(reference);
    console.log(details, teamMembers.members.length);
  }
}
```

**screenshots**:
Results are printed as belows, furthur
image

## Attestation Checklist

- [ ] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### Additional information

> Feel free to add other relevant information below

### CODE REVIEW

**Comment by corinagum** [comment]
could you also add unit tests for the new code? thanks!

**Comment by yiqing-zhao** [comment]
> could you also add unit tests for the new code? thanks!

Thanks @corinagum, some unit tests are added for the codes. Please help to review, thanks!

**Comment by corinagum** [comment]
@yiqing-zhao Thank you! Internally we are discussing some structural strategies that would affect these changes so this PR may not be merged immediately until we finalize on those decisions. We might need to iterate on your changes before merge as well, but we appreciate the contribution! I'll update this thread as more information becomes available.

**Comment by MuyangAmigo** [comment]
> @yiqing-zhao Thank you! Internally we are discussing some structural strategies that would affect these changes so this PR may not be merged immediately until we finalize on those decisions. We might need to iterate on your changes before merge as well, but we appreciate the contribution! I'll update this thread as more information becomes available.

Hey @corinagum have those discussions been finalized?

**Review by aacebo — DISMISSED** [review]

File: js/packages/teams-ai/src/Application.ts [hunk 1]
```diff
@@ -899,6 +896,106 @@ export class Application<TState extends TurnState = TurnState> {
...
+                    teamDetails = await TeamsInfo.getTeamDetails(ctx, teamId);
+                }
+            });
+        }
+
+        return teamDetails;
+    }
+
+    /**
+     * Gets a pagined list of members of one-on-one, group, or team conversation.
```
Comment (corinagum) [inline js/packages/teams-ai/src/Application.ts:964]:
```suggestion
     * Gets a paginated list of members of one-on-one, group, or team conversation.
```

File: js/packages/teams-ai/src/Application.ts [hunk 2]
```diff
@@ -899,6 +896,106 @@ export class Application<TState extends TurnState = TurnState> {
...
+     * @param context - The context for the current turn with the user.
+     * @param {number} pageSize - Suggested number of entries on a page. Page size less than 50, are treated as 50, and greater than 500, are capped at 500.
```
Comment (corinagum) [inline js/packages/teams-ai/src/Application.ts:966]:
```suggestion
     * @param {number} pageSize - Suggested number of entries on a page. Page sizes with less than 50 are treated as 50, and greater than 500, are capped at 500.
```

**Review by corinagum — DISMISSED** [review]

**Review by corinagum — APPROVED** [review]

**Review by aacebo — APPROVED** [review]

**Pull request merged**